### ISSUE TITLE
TypeError: in method 'delete_Pixmap', argument 1 of type 'struct Pixmap *'

### ISSUE THREAD

**Issue by stefan6419846** [pymupdf/PyMuPDF#1510]
## Describe the bug

PyMuPDF fails with an exception which cannot be caught or traced back to its origin:

```python
Exception ignored in: <object repr() failed>
Traceback (most recent call last):
  File "/home/user/lib/python3.6/site-packages/fitz/fitz.py", line 6977, in __del__
    self.__swig_destroy__(self)
TypeError: in method 'delete_Pixmap', argument 1 of type 'struct Pixmap *'
```

This seems to be related to a broken(?) stencil mask:

```python
Traceback (most recent call last):
  File "/home/user/tmp/test.py", line 47, in test_file
    pixmap2 = fitz.Pixmap(document, stencil_mask)
  File "/home/user/lib/python3.6/site-packages/fitz/fitz.py", line 6546, in __init__
    this = _fitz.new_Pixmap(*args)
RuntimeError: not an image
```

## To Reproduce

In my case, I am trying to extract the images embedded inside a PDF file from within a Python *unittest.TestCase*:

```python
        import fitz

        file_path = '/home/user/tmp/test_dataset/file.pdf'
        document = fitz.Document(file_path)
        for page_number in range(document.page_count):
            for image_index, image in enumerate(document.get_page_images(page_number)):
                xref = image[0]
                image_dict = document.extract_image(xref)
                if not image_dict:
                    continue

                xref = image[0]
                stencil_mask = image[1]
                print(xref)

                pixmap = fitz.Pixmap(document, xref)
                if stencil_mask > 0:
                    pixmap1 = fitz.Pixmap(pixmap)
                    try:
                        # Move the following line out of the exception handling
                        # to get the second traceback in the previous section.
                        pixmap2 = fitz.Pixmap(document, stencil_mask)
                        pixmap1.setAlpha(pixmap2.samples)
                    except RuntimeError as error:
                        print(error)
                    else:
                        pixmap = pixmap1

                if pixmap.colorspace:
                    pixmap = fitz.Pixmap(fitz.csRGB, pixmap)
                pixmap.save(f'/home/user/tmp/output_{xref}.png')
```

Plain MuPDF does not seem to fail, although `image-0032.png` is mirrored horizontally:

```
user@host:~/tmp/test_dataset> ~/tmp/mupdf-1.19.0-source/build/release/mutool info file.pdf 
file.pdf:

PDF-1.4
Info object (37 0 R):
<</Creator(TextMaker)/CreationDate(D:20200531120726)/Title<FEFF0041006E0072006500640065>/Author<FEFF004300610072006F006C0061>/Producer(TextMaker)>>
Pages: 1

Retrieving info from pages 1-1...
Mediaboxes (1):
	1	(34 0 R):	[ 0 0 595.3 841.9 ]

Fonts (4):
	1	(34 0 R):	Type0 'ComicSansMS' Identity-H (15 0 R)
	1	(34 0 R):	TrueType 'ComicSansMS' WinAnsiEncoding (19 0 R)
	1	(34 0 R):	Type0 'ComicSansMS-Bold' Identity-H (25 0 R)
	1	(34 0 R):	TrueType 'ComicSansMS-Bold' WinAnsiEncoding (29 0 R)

Images (1):
	1	(34 0 R):	[ Flate ] 731x262 8bpc DevRGB (32 0 R)

user@host:~/tmp/test_dataset> ~/tmp/mupdf-1.19.0-source/build/release/mutool extract file.pdf
extracting font-0012.ttf
extracting font-0013.ttf
extracting font-0022.ttf
extracting font-0023.ttf
extracting image-0032.png
```

The PDF file has been generated by another person. As it contains confidential information, I cannot share it here.

## Expected behavior

PyMuPDF behaves the same as MuPDF and does not error out. In case of an error, it can be caught as usual to further work with it.

## Your configuration
 - OpenSUSE Leap 15.3, x86_64
 - Python 3.6, 64 bit
 - PyMuPDF 1.19.4, statically linked with MuPDF 1.19.0-rc2, generated from source.

**Comment by JorjMcKie** [comment]
Please provide me with some reproducer, otherwise there is no way to accept the bug.
If you have confidentiality concerns, you can send the file to my e-mail.

I am not sure if a can follow every piece of your code, but why don't you do this:
```python
imgpix = fitz.Pixmap(doc, xref)
if smask > 0:
    try:
        mask = fitz.Pixmap(doc, smask)
        newpix = fitz.Pixmap(imgpix, mask)  # builds the transparent original image
        del imgpix  # recover some storage
        del mask  # recover some storage
        imgpix = newpix
    except:
        pass
# continue with imgpix ...
```

**Comment by JorjMcKie** [comment]
In the usual circumstances, you do not need to use `pix.set_alpha()`.
"Normal" here means: `imgpix` has no alpha.
If there is `imgpix.alpha == True` then currently `set_alpha` is still required - but this almost never happens.

**Comment by JorjMcKie** [comment]
If you confirm with mutool, use option `-a` which lets mutool combine each image with its mask to make the original image.

**Comment by stefan6419846** [comment]
If I recall correctly, the approach in my code has mostly been copied from the image extraction examples some months ago, which included stencil mask handling (in an additional FAQ section) like in my code. Introducing your changes still raises the same exception:

```python
                pixmap = fitz.Pixmap(document, xref)
                if stencil_mask > 0:
                    try:
                        pix_mask = fitz.Pixmap(document, stencil_mask)
                    except RuntimeError as error:
                        print(error)
                    else:
                        pix_combined = fitz.Pixmap(pixmap, pix_mask)
                        del pix_mask
                        del pixmap
                        pixmap = pix_combined
```

The code is used to process lots of different PDFs, some of them being partially broken or unreadable by other viewers, while trying to extract as much information as possible. Therefore exception handling is limited to single lines which raised issues in the past, while the remaining code worked without issues.

Using `mutool -a file.pdf` still works without issues.

**Comment by JorjMcKie** [comment]
The mentioned feature is new in version 1.19.x
So please let me have a reproducer.

**Comment by JorjMcKie** [comment]
I received it - thank you, landed in the Junk first.
I'll keep you posted.

**Comment by JorjMcKie** [comment]
The PDF is ill-defined:
The base image at xref 32 contains an image with an `/SMask` entry:
```python
print(doc.xref_object(32)
<<
  /Type /XObject
  /Subtype /Image
  /Width 731
  /Height 262
  /ColorSpace /DeviceRGB
  /SMask 30 0 R   % <== must be an image!
  /BitsPerComponent 8
  /Filter /FlateDecode
  /Length 33 0 R
>>
```
However, the definition for xref 30 is
```
<<
  /Type /XObject
  /Width 731
  /Height 262
  /Subtype /Alpha  % must be "/Image", not /Alpha!
  /ColorSpace /DeviceGray
  /BitsPerComponent 8
  /Filter /FlateDecode
  /Length 31 0 R
>>
```
This is why mutool silently ignores it (their resulting PNG is not transparent).
Whereas PyMuPDF correctly reports a "not an image" condition.
Otherwise you are right, the Python level of the Pixmap has been created, but the C level has not. This is not checked before trying to remove the resources in MuPDF (which haven't been created in this case).

**Comment by JorjMcKie** [comment]
I have a new version. But I saw you generated from source, so a pre-version wheel won't help you presumably.
Please let me know otherwise.

**Comment by JorjMcKie** [comment]
Fixed in v1.19.5 currently being uploaded.
